calicoctl/commands/node/run.go

```diff
@@ -0,0 +1,312 @@
+	"github.com/projectcalico/calico-containers/calicoctl/commands/clientmgr"
+	"github.com/projectcalico/calico-containers/calicoctl/commands/docoptutils"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/etcd"
+)
+
+const (
+	ETCD_KEY_NODE_FILE     = "/etc/calico/certs/key.pem"
+	ETCD_CERT_NODE_FILE    = "/etc/calico/certs/cert.crt"
+	ETCD_CA_CERT_NODE_FILE = "/etc/calico/certs/ca_cert.crt"
+)
+
+// Run function collects diagnostic information and logs
+func Run(args []string) {
+	var err error
+	doc := `Usage:
+  calicoctl node run [--ip=<IP>] [--ip6=<IP6>] [--as=<AS_NUM>]
+                     [--name=<NAME>]
+                     [--log-dir=<LOG_DIR>]
+                     [--node-image=<DOCKER_IMAGE_NAME>]
+                     [--backend=<BACKEND>]
+                     [--config=<CONFIG>]
+                     [--no-default-ippools]
+                     [--dryrun]
+
+Options:
+  -h --help                Show this screen.
+     --as=<AS_NUM>         The default AS number for this node.  If this is not
+                           specified, the node will use the global AS number
+                           (see 'calicoctl config' for details).
+     --name=<NAME>         The name of the Calico node.  If this is not
+                           supplied it defaults to the host name.
+     --ip=<IP>             The local management address to use.  If this is not
+                           specified, the node will attempt to auto-discover
+                           the local IP address to use - however, it is
+                           recommended to specify the required address to use.
+     --ip6=<IP6>           The local IPv6 management address to use.  If this
+                           is not specified, the node will not route IPv6.
+     --log-dir=<LOG_DIR>   The directory containing Calico logs.
+                           [default: /var/log/calico]
+     --node-image=<DOCKER_IMAGE_NAME>
+                           Docker image to use for Calico's
+                           per-node container.
+                           [default: calico/node:latest]
+     --backend=<BACKEND>   Specify which networking backend to use.  Choices
+                           are "bird", "gobgp" or "none". When set to "none",
+                           Calico node run in policy only mode.
+                           [default: bird]
+     --dryrun              Output the appropriate Docker command, without
+                           starting the container.
+     --no-default-ippools  Do not create default pools upon startup.  If this
+                           is not set, and there are no Calico IP pools already
+                           created, then a default IP pool will be created.
+  -c --config=<CONFIG>     Filename containing connection configuration in
+                           YAML or JSON format.
+                           [default: /etc/calico/calicoctl.cfg]
```

Comment:
If the file does not exist then we load from environment.  This is as per all the other commands.